src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -31,5 +31,40 @@ internal class ConnectionHandlerContext : HandlerRequest
         /// Class used by orchestrator while chaining handlers.
         /// </summary>
         public Exception Error { get; set; }
+
+        /// <summary>
+        /// Method needed to clone ConnectionHandlerContext as part of history along the chain.
+        /// </summary>
+        public object Clone() 
+        {
+            ConnectionHandlerContext clonedContext = new ConnectionHandlerContext();
+
+            if (this.ConnectionStream != null)
+            {
+                MemoryStream clonedStream = new MemoryStream();
+                this.ConnectionStream.CopyTo(clonedStream);
```

Comment:
This will copy content of `this.ConnectionStream` starting from its current `Position` and move its position to its end. Is it what you intended?